libs/azure-storage/langchain_azure_storage/document_loaders.py

```diff
@@ -253,7 +253,9 @@ def _yield_blob_names(self, container_client: ContainerClient) -> Iterator[str]:
         if self._blob_names is not None:
             yield from self._blob_names
         else:
-            yield from container_client.list_blob_names(name_starts_with=self._prefix)
+            for blob in container_client.list_blobs(name_starts_with=self._prefix):
+                if blob.size > 0:
```

Comment:
When we go to add the additional `hdi_isfolder` check, let's make sure to break this down into a helper method `_is_adls_directory()` to help with readability.